### PR TITLE
Initial draft of Drain mode feature

### PR DESCRIPTION
**Pull request by GaneshSPatil**
screen shot 2019-01-02 at 2 45 38 pm

### CODE REVIEW

File: advanced_usage/drain_mode.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# **Drain Mode**
+
+**Note: **Using this feature requires atleast version `19.1.0`  of the GoCD server.
+
+The GoCD system includes a lot of internal subsystems and processes, which continuously do work. 
+Some examples include the material subsystem which takes care of polling for materials, the scheduling subsystem which automatically starts pipelines and stages, the agent subsystem which continuously keeps tracks of agent states and provides them jobs to run, etc.
+
+The drain mode is a maintenance mode which a GoCD system administrator can put GoCD into so that it is safe to restart it or upgrade it without having running jobs reschedule when it is back.
```
Comment (adityasood) [inline advanced_usage/drain_mode.md]: I think we might have to rephrase this to something like 

"Drain mode is a maintenance mode which can be utilized by the GoCD system administrator to safely restart or upgrade the GoCD server without having any running jobs reschedule when the server is back up."

File: advanced_usage/drain_mode.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - Shows state of materials which are being polled currently.
+  - Shows state of agents which are currently running jobs.
+  - Shows a banner near the footer of all pages which tells everyone that the server is in drain mode.
+  - Indicate when a server is safe to restart or upgrade (finished draining).
+
+[Ganesh will add screenshots here]
+
+**NOTE: ** `Upon a restart, the GoCD server will no longer be in drain mode and will start all subsystems.`
+
+The [drain mode API](https://api.gocd.org/current/#drain_mode) allows users to enable, disable and monitor the server drain mode state.
```
Comment (adityasood) [inline advanced_usage/drain_mode.md]: allows GoCD administrators would be apt

**Comment by GaneshSPatil** [comment]
#### Pending things:
  - [ ] Add screenshots
  - [ ] Provide links to mdu, build assignments, etc
